core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java

```diff
@@ -675,4 +686,103 @@ public String getResourceCrnByResourceName(String resourceName) {
     public EnumSet<Crn.ResourceType> getSupportedCrnResourceTypes() {
         return EnumSet.of(Crn.ResourceType.IMAGE_CATALOG);
     }
+
+    private StatedImage getStatedImage(long workspaceId, String catalogName, String imageId)
```

Comment:
This class is too complex as it is without our changes as well. I'd move out our custom image implementation into a separated component (as it was discussed during the handover)

Reply:
The possible parts were moved to the CustomImageProvider class.